src/portable/portable.c

```diff
@@ -1351,15 +1358,22 @@ static int attach_unit_file(
 
         if ((flags & PORTABLE_PREFER_SYMLINK) && m->source) {
 
+                if (flags & PORTABLE_FORCE_ATTACH)
+                        (void) unlink(path);
+
                 if (symlink(m->source, path) < 0)
```

Comment:
i think in cases like this it might make sense to use symlink_atomic() rather than unlink()+symlink() when replacing. Not because it's atomic behaviour (which is pretty though, and doesn't hurt), but mostly because it has a nicer failure mode: it will create a symlink first (and thus make sure the operation can succeed at all) before destroying anything.

(here and above)